astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -60,12 +50,38 @@ def __init__(  # nosec: B107
         self.port = port
         self.username = username
         self.password = password
-        self.known_hosts = known_hosts
+        self.known_hosts: bytes | str = os.path.expanduser(known_hosts)
         self.key_file = key_file
         self.passphrase = passphrase
         self.private_key = private_key
-        self.no_host_key_check = True
-        self.host_key = None
+
+    def _parse_extras(self, conn: Connection) -> None:
+        """Parse extra fields from the connection into instance fields"""
+        extra_options = conn.extra_dejson
+        if "key_file" in extra_options and self.key_file == "":
+            self.key_file = extra_options["key_file"]
+        if "known_hosts" in extra_options:
```

Comment:
Only take extra if value not explicitly passed like we do with key file?

```suggestion
        if "known_hosts" in extra_options and self.known_hosts != self.default_known_hosts:
```